tests/test_figure_processing.py

```diff
@@ -69,6 +69,50 @@ def test_myst_image(self):
             '<figure class="informal"><img alt="flower" ' +
             'src="_images/flower.png"/></figure>')
 
+    def test_extra_p_tags_and_spaces_are_removed_from_captions(self):
+        """
+        In order to render captions correctly, we should ensure
+        that the <figcaption> contains only text, without any extraneous
+        whitespace or <p> tags
+        """
+        text = """<figure class="align-default" id="example-fig">
+<a class="reference internal image-reference" href="images/flower.png">
+<img alt="images/flower.png" src="../_images/flower.png"
+style="height: 150px;" /></a>
+<figcaption>
+<p><span class="caption-number">Fig. 1 </span><span class="caption-text">
+Here is my figure caption!</span>
+<a class="headerlink" href="#example-fig" title="Permalink to this image">#</a>
+</p></figcaption>"""
+        chapter = Soup(text, 'html.parser')
+        result = process_figures(chapter, Path('example'))
+        caption = result.find("figcaption")
+        assert not caption.p
+        assert "\n" not in caption.string
+        assert caption.string[0] != " "
+
+    @pytest.mark.in_dev
+    def test_markup_is_preserved_in_captions(self):
+        """
+        Any markup in a figure caption should be preserved, but there
+        should be no left spacing or newlines still
+        """
+        text = """<figure class="align-default" id="example-fig">
+<a class="reference internal image-reference" href="images/flower.png">
+<img alt="images/flower.png" src="../_images/flower.png"
+style="height: 150px;" /></a>
+<figcaption>
+<p><span class="caption-number">Fig. 1 </span><span class="caption-text">
+Here is my <code>figure</code> caption!</span>
+<a class="headerlink" href="#example-fig" title="Permalink to this image">#</a>
+</p></figcaption>"""
+        chapter = Soup(text, 'html.parser')
+        result = process_figures(chapter, Path('example'))
+        caption = result.find("figcaption")
+        assert not caption.p
+        assert caption.find("code")
```

Comment:
Idea (non-blocking): Could add a `<strong>` in the test caption for some extra coverage.

Reply:
That's a good idea; instead though I'll add a test to ensure that if there is formatting at the _beginning_ of the caption, we don't magically lose spaces in the rest of the caption. I'll use a `<strong>` tag there.

Reply:
Okay, see https://github.com/oreillymedia/jupyter-book-to-htmlbook/pull/33/commits/69950444b3066460ab40b156f08de01d9cfa635d